yt/convenience.py

```diff
@@ -12,112 +12,114 @@
-                    valid_file.append(True)
-                    args[argno] = os.path.join(ytcfg.get("yt", "test_data_dir"), arg)
-                else:
-                    valid_file.append(False)
-        else:
-            valid_file.append(False)
-    types_to_check = output_type_registry
-    if not any(valid_file):
-        try:
-            from yt.data_objects.time_series import DatasetSeries
-            ts = DatasetSeries.from_filenames(*args, **kwargs)
-            return ts
-        except (TypeError, YTOutputNotIdentified):
-            pass
-        # We check if either the first argument is a dict or list, in which
-        # case we try identifying candidates.
-        if len(args) > 0 and isinstance(args[0], (list, dict)):
-            # This fixes issues where it is assumed the first argument is a
-            # file
-            types_to_check = dict((n, v) for n, v in
-                    output_type_registry.items() if n.startswith("stream_"))
-            # Better way to do this is to override the output_type_registry
+    fn = os.path.expanduser(fn)
+
+    if any([wildcard in fn for wildcard in "[]?!*"]):
+        from yt.data_objects.time_series import DatasetSeries
+        return DatasetSeries(fn, *args, **kwargs)
+
+    if not (os.path.exists(fn) or fn.startswith("http")):
+        test_path = os.path.join(ytcfg.get("yt", "test_data_dir"), fn)
+        if os.path.exists(test_path):
+            fn = test_path
         else:
-            mylog.error("None of the arguments provided to load() is a valid file")
-            mylog.error("Please check that you have used a correct path")
-            raise YTOutputNotIdentified(args, kwargs)
-    for n, c in types_to_check.items():
-        if n is None: continue
-        if c._is_valid(*args, **kwargs): candidates.append(n)
-
-    # convert to classes
-    candidates = [output_type_registry[c] for c in candidates]
+            raise IOError("No such file or directory: %s" % fn
+
+    candidates = []
+    for n, c in output_type_registry.items():
+        if n is not None and c._is_valid(fn, *args, **kwargs):
+            candidates.append(c)
+
     # Find only the lowest subclasses, i.e. most specialised front ends
     candidates = find_lowest_subclasses(candidates)
+
     if len(candidates) == 1:
-        return candidates[0](*args, **kwargs)
+        return candidates[0](fn, *args, **kwargs)
     if len(candidates) == 0:
         if ytcfg.get("yt", "enzo_db") != '' \
-           and len(args) == 1 \
-           and isinstance(args[0], str):
+           and len(args) == 0 \
+           and isinstance(fn, str):
             erdb = EnzoRunDatabase()
-            fn = erdb.find_uuid(args[0])
+            fn = erdb.find_uuid(fn)
             n = "EnzoDataset"
             if n in output_type_registry \
                and output_type_registry[n]._is_valid(fn):
                 return output_type_registry[n](fn)
```

Comment:
This special case looks dead to me: it is neither hit by the tests or documented anywhere.
I'd love to get feedback on this ! Could it also be cut out into its own function ?

Reply:
I'm cutting this in a clearly identified commit. I'll resurrect it if needed.

Reply:
Since it seems impossible to know in advance if this is still in use, I reintroduced it as a separate function `load_enzo_db()`. If accepted in this state, this will need to be clearly documented in the release notes.